clients/src/main/java/org/apache/kafka/common/requests/MetadataRequest.java

```diff
@@ -52,22 +70,33 @@ public AbstractRequestResponse getErrorResponse(int versionId, Throwable e) {
         Errors error = Errors.forException(e);
         List<MetadataResponse.PartitionMetadata> partitions = Collections.emptyList();
 
-        for (String topic : topics)
-            topicMetadatas.add(new MetadataResponse.TopicMetadata(error, topic, partitions));
+        if (topics != null) {
+            for (String topic : topics)
+                topicMetadatas.add(new MetadataResponse.TopicMetadata(error, topic, false, partitions));
+        }
 
         switch (versionId) {
             case 0:
-                return new MetadataResponse(Collections.<Node>emptyList(), topicMetadatas);
+            case 1:
```

Comment:
I don't think we can do a fall-through here? Since v1 error response is not a valid v0 response?